### ISSUE TITLE
Create/update components in preparation for collections/resources file hierarchy

### ISSUE THREAD

**Issue by kwajiehao** [isomerpages/isomercms-frontend#173]
## Overview
We agreed that the current layout for pages and resource rooms does not make sense: larger repositories like MLAW might have hundreds of pages and resources - displaying them all in a single list would make it hard to discover or find a single document. It would make a lot more sense to adopt a _hierarchical_ style where collections/categories are displayed first, and only after you click into a particular collection/category will the pages contained in it be displayed. This would be more natural to users (since most computers and Google Drive adopt this hierarchical file system) and cleaner from a design perspective.

This issue identifies the components that we need to update to achieve this hierarchical file system. There are three main components:
- Collection/category card
- Page/resource card
- Page/resource settings modal

### Collection/category card

<img width="997" alt="Screenshot 2020-10-26 at 2 13 00 PM" src="https://user-images.githubusercontent.com/31984694/97140011-c0ac9080-1796-11eb-87e4-af552f481963.png">

A simple card will do. We can ignore the settings modal which allows you to "Edit Details" or "Delete" a collection/category, since this is something we can implement later. This card is to be used for both collections and resource categories. Upon clicking on this card, all pages in a collection, or all resources in a resource category should be displayed.

### Page/resource card

<img width="419" alt="Screenshot 2020-10-26 at 2 14 08 PM" src="https://user-images.githubusercontent.com/31984694/97140255-239e2780-1797-11eb-8626-bb143e256471.png">

For this card, we want to use the existing resource card as a reference. There should be three types of resource/page cards:
1. A base card (without either date - highlighted in orange - or collection/category name - highlighted in red). This will be used by pages that are not in a collection
2. A collection card with the collection name (highlighted in red) but without the date. This will be used by collection pages
3. A resource card with both collection name and date (pictured). This will be used by resources.

These can be implemented as HOCs over each preceding component.

A minor fix we will need to make is to place the date on the same level as the settings gear.

We will implement the settings modal (pictured below) in a separate issue.

<img width="879" alt="Screenshot 2020-10-26 at 2 13 36 PM" src="https://user-images.githubusercontent.com/31984694/97140438-927b8080-1797-11eb-84d7-0151a643beab.png">

### Page/resource settings modal

<img width="696" alt="Screenshot 2020-10-26 at 2 12 32 PM" src="https://user-images.githubusercontent.com/31984694/97140498-b6d75d00-1797-11eb-8486-393a65c7f9eb.png">

There are two types of page/resource settings modal we want:

1. Page settings modal - just the part highlighted in red. A couple of scenarios to take note of:
- When a user is within a collection and trying to create a new page, the **collection field will be disabled** - we do not want to allow them to choose a collection different from the one they're already in. 
- When the user clicks on the settings for an existing collection page, the **collection field will be disabled**. Pages should be moved from one collection to another using a different mechanism.
- When the user is on the pages layout (only collections are displayed) and tries to create a page, the collections field should be a dropdown displaying existing collections (to be passed in as props). **We will not allow users to create a new collection for the time being** - this is a feature that we can implement for a separate issue. 

2. Resource settings modal - the entire pictured modal. The same scenarios that apply to the collections field in part 1 will apply here as well, for resource categories.

**Comment by alexanderleegs** [comment]
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/184
